pyartcd/pyartcd/pipelines/doomsday.py

```diff
@@ -0,0 +1,74 @@
+import os
+from typing import Optional
+import click
+from time import sleep
+import shutil
+
+from pyartcd.runtime import Runtime
+from pyartcd.cli import cli, pass_runtime
+from artcommonlib.exectools import cmd_assert
+from doozerlib.util import mkdirs
+
+
+class DoomsdaySync:
+    ALL_ARCHES_LIST = ["x86_64", "s390x", "ppc64le", "aarch64", "multi"]
+    N_RETRIES = 4
+
+    def __init__(self, runtime: Runtime, all_arches: bool, arches: Optional[list[str]], version: str):
+        self.runtime = runtime
+        self.version = version
+        self.workdir = "./workspace"
+
+        if arches and all_arches:
+            raise Exception("Cannot specify both --arches and --all-arches")
+
+        if not (arches or all_arches):
+            raise Exception("Either --arches or --all-arches needs to be specified")
+
+        self.arches = arches.split(",") if not all_arches else self.ALL_ARCHES_LIST
+
+    def sync_arch(self, arch: str):
+        if arch not in self.ALL_ARCHES_LIST:
+            raise Exception(f"Invalid arch: {arch}")
+
+        major_minor = ".".join(self.version.split(".")[:2])
+        path = f"{self.workdir}/{major_minor}/{self.version}/{arch}"
+        mirror_cmd = f"oc adm release mirror quay.io/openshift-release-dev/ocp-release:{self.version}-{arch} --keep-manifest-list --to-dir={path}"
+        aws_cmd = f"aws s3 sync {path} s3://ocp-doomsday-registry/release-image/{path}"
```

Comment:
Should `workdir` be part of the bucket?

Reply:
`cmd_assert` accepts both a string and a list as command. Think list is preferred. This allows python to do the shell escaping for us. 

```
mirror_cmd = ["oc', 'adm', 'release', 'mirror',
   f'quay.io/openshift-release-dev/ocp-release:{self.version}-{arch}', 
  '--keep-manifest-list ',
  '--to-dir={path}']
```

Reply:
> Should workdir be part of the bucket?

Thanks for pointing that out, complete missed it. Should be fixed now
> cmd_assert accepts both a string and a list as command. Think list is preferred. This allows python to do the shell escaping for us.

Also fixed